hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/OAuth2AuthProvider.java

```diff
@@ -0,0 +1,65 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http;
+
+import pl.allegro.tech.hermes.api.OAuth2AuthenticationData;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.common.cache.zookeeper.NodeCache;
+import pl.allegro.tech.hermes.consumers.consumer.ConsumerMessageSender;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.core.model.OAuthConfig;
+
+public class OAuth2AuthProvider implements HttpAuthorizationProvider {
+
+    private final OAuth2AuthenticationData authData;
+
+    private final OAuth20Service service;
+
+    private String accessToken;
+
+    public OAuth2AuthProvider(Subscription subscription) {
+        this.authData = subscription.getOAuth2AuthenticationData();
+
+        service = new ServiceBuilder()
+                .apiKey(authData.getConsumerKey())
+                .apiSecret(authData.getConsumerSecret())
+                .build(new OAuthApi(authData.getAccessTokenEndpoint()));
+        obtainAccessToken();
```

Comment:
You can't make online calls from this constructor. Have you checked which thread is running this code and when? You can only make online and blocking calls on consumer thread (if any blocking calls at all).